gossip/group.go

```diff
@@ -124,7 +124,10 @@ func (g *group) compact() bool {
 	g.minTTLStamp = math.MaxInt64
 	g.gatekeeper = nil
 	for key, i := range g.Infos {
-		if i.TTLStamp <= now {
+		if i.expired(now) {
+			// We do not need to call g.removeInternal()
+			// since we will soon add a new info that
+			// becomes a new gatekeeper.
```

Comment:
I'd rephrase as: `We do not need to call g.removeInternal() since we've already cleared the gatekeeper and minTTLStamp`. That said, I don't think there is any harm in calling `removeInternal` here and doing so localizes the deletion from the `Infos` map to `removeInternal`.